src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidContentProviderBroker.cs

```diff
@@ -0,0 +1,393 @@
+                {
+                    _logger.Warning("Android broker is either not installed or not reachable so no accounts will be removed. ");
+                    return;
+                }
+
+                await PerformRemoveAccountsAsync(applicationConfiguration, account).ConfigureAwait(false);
+            }
+        }
+
+        private async Task PerformRemoveAccountsAsync(ApplicationConfiguration applicationConfiguration, IAccount account)
+        {
+            try
+            {
+                InitiateBrokerHandShakeAsync();
+
+                await RemoveBrokerAccountFromBrokersAsync(applicationConfiguration.ClientId, account).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error("Failed to remove Android broker account from the broker. ");
+                _brokerHelper.HandleBrokerOperationError(ex);
+                throw;
+            }
+        }
+
+        private async Task RemoveBrokerAccountFromBrokersAsync(string clientId, IAccount account)
+        {
+            var removeAccountsBundle = _brokerHelper.CreateRemoveBrokerAccountBundle(clientId, account);
+            var operationBundleJSON = SearializeBundleToJSON(removeAccountsBundle);
+            await PerformContentResolverOperationAsync(ContentResolverOperation.removeAccounts, operationBundleJSON).ConfigureAwait(false);
+        }
+
+        private async Task<Bundle> PerformContentResolverOperationAsync(ContentResolverOperation operation, string OperationParameters)
+        {
+            ContentResolver resolver = GetContentResolver();
+
+            ICursor resultCursor = null;
+            await Task.Run(() => resultCursor = resolver.Query(AndroidUri.Parse(GetContentProviderURIForOperation(Enum.GetName(typeof(ContentResolverOperation), operation))),
+                                                            !string.IsNullOrEmpty(_negotiatedBrokerProtocolKey) ? new string[] { _negotiatedBrokerProtocolKey } : null,
+                                                            OperationParameters,
+                                                            null,
+                                                            null)).ConfigureAwait(false);
+
+            if (resultCursor == null)
+            {
+                _logger.Error("An error occurred during the content provider operation.");
+                throw new MsalClientException(MsalError.CannotInvokeBroker, "Could not communicate with broker via content provider.");
+            }
+
+            var resultBundle = resultCursor.Extras;
+            resultCursor.Close();
+
+            return resultBundle;
+        }
+
+        private ContentResolver GetContentResolver()
+        {
+            if (_parentActivity == null)
+            {
+                return Application.Context.ContentResolver;
+            }
+            else
+            {
+                return _parentActivity.ContentResolver;
+            }
```

Comment:
nit: could be turned into: `return _parentActivity?.ContentResolver ?? Application.Context.ContentResolver`